tests/integration/security/authorization_test.go

```diff
@@ -1037,80 +1043,66 @@ func TestAuthorization_Path(t *testing.T) {
 	framework.NewTest(t).
 		Features("security.authorization.path-normalization").
 		Run(func(ctx framework.TestContext) {
-			ns := namespace.NewOrFail(t, ctx, namespace.Config{
-				Prefix: "v1beta1-path",
-				Inject: true,
-			})
-			ports := []echo.Port{
-				{
-					Name:        "http",
-					Protocol:    protocol.HTTP,
-					ServicePort: 80,
-					// We use a port > 1024 to not require root
-					InstancePort: 8090,
-				},
-			}
-
-			var a, b echo.Instance
-			echoboot.NewBuilder(ctx).
-				With(&a, echo.Config{
-					Service:   "a",
-					Namespace: ns,
-					Subsets:   []echo.SubsetConfig{{}},
-					Ports:     ports,
-				}).
-				With(&b, echo.Config{
-					Service:   "b",
-					Namespace: ns,
-					Subsets:   []echo.SubsetConfig{{}},
-					Ports:     ports,
-				}).
-				BuildOrFail(t)
-
-			newTestCase := func(path string, expectAllowed bool) rbacUtil.TestCase {
-				return rbacUtil.TestCase{
-					Request: connection.Checker{
-						From: b,
-						Options: echo.CallOptions{
-							Target:   a,
-							PortName: "http",
-							Scheme:   scheme.HTTP,
-							Path:     path,
-						},
-					},
-					ExpectAllowed: expectAllowed,
-				}
-			}
-			cases := []rbacUtil.TestCase{
-				newTestCase("/public", true),
-				newTestCase("/private", false),
-				newTestCase("/public/../private", false),
-				newTestCase("/public/./../private", false),
-				newTestCase("/public/.././private", false),
-				newTestCase("/public/%2E%2E/private", false),
-				newTestCase("/public/%2e%2e/private", false),
-				newTestCase("/public/%2E/%2E%2E/private", false),
-				newTestCase("/public/%2e/%2e%2e/private", false),
-				newTestCase("/public/%2E%2E/%2E/private", false),
-				newTestCase("/public/%2e%2e/%2e/private", false),
-			}
-
+			ns := apps.Namespace3
 			args := map[string]string{
 				"Namespace": ns.Name(),
 			}
 			policies := tmpl.EvaluateAllOrFail(t, args,
 				file.AsStringOrFail(t, "testdata/authz/v1beta1-path.yaml.tmpl"))
 			ctx.Config().ApplyYAMLOrFail(t, ns.Name(), policies...)
 			defer ctx.Config().DeleteYAMLOrFail(t, ns.Name(), policies...)
-
-			rbacUtil.RunRBACTest(t, cases)
+			for _, srcCluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("In %s", srcCluster.Name())).Run(func(ctx framework.TestContext) {
+					f := apps.F.Match(echo.InCluster(srcCluster).And(echo.NameSpace(ns.Name())))
```

Comment:
why do we need to use some new app (`f` and `e`) in this test case? Anything special for this test case?

Reply:
The echo configuration looks a little bit different than current `a` `b` set up. https://github.com/istio/istio/pull/28088/files#diff-84045abbcab1322dfe6752d85763aee916a8811a98783438cdae5d0e4a05aac7R165-R173. Thus I renamed it to `f` and `e` to keep the original setup.

Reply:
`a` and `b` are good if the apps are very simple and don't have anything special. If the apps have special config, try naming them something descriptive.